tsdb/head.go

```diff
@@ -1738,6 +1738,7 @@ type memSeries struct {
 	ref          uint64
 	lset         labels.Labels
 	chunks       []*memChunk
+	chunksLock   sync.RWMutex
```

Comment:
The `memSeries` itself is a `Mutex` which is held before calling `truncateChunksBefore` and `append`. Can we change it to `RWMutex` and use that instead?

Reply:
sure thing, done :)